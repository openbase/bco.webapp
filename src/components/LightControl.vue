<template>
  <v-container>
    <v-layout text-xs-center wrap row justify-center>
      <v-flex mb-4>
        <h1 class="display-2 font-weight-bold mb-3">
          Light Control Component
        </h1>
        <p class="subheading font-weight-regular">
          This component helps you to visualize the current state
          <br />and to switch it on or off.
        </p>
      </v-flex>

      <v-flex xs12>
        <p v-if="switch1">
          <v-icon x-large color="green darken-2">mdi-lightbulb</v-icon>
        </p>
        <p v-else>
          <v-icon x-large color="red darken-2">mdi-lightbulb</v-icon>
        </p>
      </v-flex>

      <v-flex xs12 md2>
        <v-switch
          v-model="switch1"
          :label="`Switch 1: ${switch1.toString()}`"
          @change="postLightSwitch()"
        ></v-switch>
      </v-flex>
    </v-layout>
  </v-container>
</template>
<script>
import axios from "axios";

export default {
  data: function() {
    return {
      switch1: true
    };
  },
  created() {
    axios
      .get(`http://scummbar:8484/light`)
      .then(response => {
        this.switch1 = response.data.power_state.value === "ON";
      })
      .catch(e => {
        console.error(JSON.stringify(e, null, 2));
      });
  },
  methods: {
    postLightSwitch() {
      const value = this.switch1 ? "ON" : "OFF";

      axios
        .post(`http://scummbar:8484/light`, {
          id: "53b59c91-dd89-4a24-95ae-0ba841634039",
          power_state: {
            value
          }
        })
        .catch(e => {
          console.error(JSON.stringify(e, null, 2));
        });
    }
  }
};
</script>

<style></style>
