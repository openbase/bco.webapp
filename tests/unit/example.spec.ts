import { shallowMount } from "@vue/test-utils";

import Vue from "vue";
import Vuetify from "vuetify";

Vue.use(Vuetify);
import LightControl from "@/components/LightControl.vue";

describe("LightControl.vue", () => {
  it("renders props.msg when passed", () => {
    const msg = "new message";
    const wrapper = shallowMount(LightControl, {
      propsData: { msg }
    });
    expect(wrapper.text()).toMatchSnapshot();
  });
});
