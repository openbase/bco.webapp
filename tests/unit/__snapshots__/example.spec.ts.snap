// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelloWorld.vue renders props.msg when passed 1`] = `
"Welcome to Vuetify
       
        For help and collaboration with other Vuetify developers,
        please join our online
        Discord Community What's next? 
          Explore components
        
          Select a layout
        
          Frequently Asked Questions
         Important Links 
          Documentation
        
          Chat
        
          Made with Vuetify
        
          Twitter
        
          Articles
         Ecosystem 
          vuetify-loader
        
          github
        
          awesome-vuetify"
`;
